lib/trino-collect/src/test/java/io/trino/collect/cache/TestEvictableCache.java

```diff
@@ -542,4 +546,42 @@ public void testInvalidateAndLoadConcurrently(Invalidation invalidation)
             executor.awaitTermination(10, SECONDS);
         }
     }
+
+    @Test(dataProvider = "disabledCacheImplementations")
+    public void testPutOnEmptyCache(DisabledCacheImplementation disabledCacheImplementation)
+    {
+        Map<Object, Object> cache = EvictableCacheBuilder.newBuilder()
+                .maximumSize(0)
+                .disabledCacheImplementation(disabledCacheImplementation)
+                .build()
+                .asMap();
```

Comment:
`cache` should be a cache, not "cache's map view"

Reply:
This is about the "cache's map view". Regular cache does not expose `put` methods that return anything.

Reply:
keep `cache` name, make it `Cache` and use `cache.asMap().put` below.
or  (below)
```
cacheMap = cache.asMap()`
```

but keep `cache` declaration as a Cache